### DISCUSSION TITLE
lazy-rules on q-input ignored after first failed validation

### DISCUSSION THREAD

**Discussion by cdecinkoKnight** [quasarframework/quasar#16181] [General - Components / Directives / etc] [answered]
I have lazy-rules set on my q-input field. I open the form, enter some text and on blur, validation occurs. That works well. If my validated text fails, I get the error message. I start to modify, say reduce characters, red error goes away, i.e. validation resets. However, now, validation occurs with every single keystroke! I still only want validation to fire on blur. If I enter an invalid value, when I start to correct it, clear the validation and then only apply validation when I leave the field.

**Comment by metalsadman** [comment]
that's kind of a weird requirement, if you want solely to validate always on blur then you can do this combo `lazy-rules
      @keydown="inputRef.resetValidation()"
      @blur="inputRef.validate()"`

**Reply by cdecinkoKnight** [reply]
I'm doing something perhaps a bit different. When the validation hits a trigger it opens a dialog. When the email already exists it asks if they want to edit the existing record. Problem is, they keep getting the dialog while typing the correct email address.

**Reply by cdecinkoKnight** [reply]
I must be daft, but what does inputRef refer to? I've tried various ways to refer to the current control but each give me an error.

**Reply by metalsadman** [reply, marked as answer]
https://codepen.io/metalsadman/pen/PoxrEGm?editors=101

**Reply by cdecinkoKnight** [reply]
Ok, that works but I'm getting this error in VSCode:  '__VLS_ctx.inputRef' is possibly 'null'

**Reply by metalsadman** [reply]
Just put optional chaining ie. `inputRef?.`

**Reply by enmaku** [reply]
That's not a weird requirement at all, and I've just run up against it as well.

Imagine you're filling out your address on a form, a key sticks and you accidentally enter a 6-digit zip code, you hit tab and are greeted with an error. Shift+tab back to the field and start typing and you now get a fresh validation error on every single keystroke, which is not a good user experience.

If quasar has a keyword that says in the documentation will cause validation only on blur, then it should *always* cause validation only on blur.

**Reply by metalsadman** [reply]
validations are opinonated, wrap, make a composble, helper function etc... then reuse.
